CSAF Vulnerability Report profile to initiate coordination of a vulnerability (#1513)
oasis-tcs/csaf: csaf_2.1/prose/edit/src/profiles.md

- The value of `$.document.category` SHALL be `csaf_superseded`.
- The elements `$.product_tree` and `$.vulnerabilities` SHALL NOT exist.

## Profile 9: Vulnerability Report

This profile SHOULD be used to exchange a vulnerability report during the Coordinated Vulnerability Disclosure (CVD) process.
It supports communication between discoverers, vendors, and, where applicable, coordinators prior to public disclosure.
A Vulnerability Report is intended for use during the private coordination process prior to public disclosure.
It provides a structured mechanism for exchanging information throughout the coordinated disclosure process among the participating parties.
Information such as affected products, technical details, severity assessments, and supporting evidence can be
structured and exchanged privately using this profile to facilitate coordination.

A CSAF document SHALL fulfill the following requirements to satisfy the profile "Vulnerability Report":

- The following elements MUST exist and be valid:
  - All elements required by the profile "CSAF Base".

> 

Reviewer: ```suggestion
```

  - `$.document.publisher`

    > Identifies the discoverer or organization submitting the report to initiate the coordination process.

  - `$.product_tree`

    > Lists all products referenced within the report.

  - `$.vulnerabilities`

    > Lists one or more reported vulnerabilities.

  - `$.vulnerabilities[*].ids`

    > Provides a unique identifier for each reported vulnerability in the absence of a public CVE ID.

>     > The publisher SHOULD assign a unique identifier for each reported vulnerability to ensure clear communication

Reviewer: Either use `$.document.publisher` or use the generic term "document issuer" / "issuing party"

```suggestion
    > The issuing party SHOULD assign a unique identifier for each reported vulnerability to ensure clear communication
```

Author: This has changed as well - may be no longer relevant.

>     > throughout the coordination process until a public identifier becomes available.

Reviewer: ```suggestion
    > throughout the coordination process until a public identifier becomes available.
    > Such unique identifier MAY be included in the released security advisory.
```

Author: This has changed significantly now.

    > Identifiers MAY use any naming convention that uniquely identifies the vulnerability within the publisher's

>     > tracking system.

Reviewer: ```suggestion
    > Identifiers MAY use any naming convention that uniquely identifies the vulnerability within the 
    > issuing party's tracking system.
```

  - `$.vulnerabilities[*].notes`

    > Provides the technical description and supporting information for each reported vulnerability.

  - `$.vulnerabilities[*].notes[*].category`

    > The `description` category SHALL be used to describe the vulnerability and the related defect or flaw.
    > The `other` category MAY be used for supporting information, such as the vulnerability discovery method,
    > testing methodology, proof-of-concept, or other information relevant to the investigation.

  - `$.vulnerabilities[*].notes[*].title`

    > The `title` field SHOULD provide a concise description of the note.
    > Examples include "Vulnerability Description", "Vulnerability Discovery Method",

>     > or "Proof-of-Concept".

Reviewer: The requirement is currently weak. We don't say that there is a specific title that we want to see. Please take a look at profile 7/8 for required notes.

  - `$.vulnerabilities[*].notes[*].audience`

    > This profile RECOMMENDS using values consistent with the actor terminology already used by
    > `$.vulnerabilities[*].involvements[*].party`.

>     > Recommended values include `discoverer`, `coordinator`, and `vendor`.

Reviewer: Not quite sure about that as it would require that every note has an `audience`. Also, what about a technical description vs a management summary? I would expect different audiences here...

  - `$.vulnerabilities[*].product_status`

    > Lists each product's status with respect to the reported vulnerability as determined by the discoverer,

>     > who is the `$.document.publisher` of this document.

Reviewer: This would essentially prevent anonymous reporting as it explains that the `$.document.publisher` is always the discoverer...
What about:

```suggestion
    > Lists each product's status with respect to the reported vulnerability as determined by the 
    > issuing party of this document.
```

  - `$.vulnerabilities[*].product_status.known_affected`

    > Identifies the products that the discoverer, acting as the `$.document.publisher`,

>     > has determined to be affected.

Reviewer: See above

  - Each vulnerability SHALL identify at least one affected product in
    `$.vulnerabilities[*].product_status.known_affected`.

    > This requirement ensures that a Vulnerability Report identifies at least one product
    > to which the reported vulnerability is known to apply.

- The value of `$.document.category` SHALL be `csaf_vulnerability_report`.

- The following elements SHOULD exist:

  - `$.document.acknowledgments`

    > Identifies the individuals or organizations that should receive public acknowledgment for the discovery.
    > If omitted, it indicates that the discoverer does not wish to receive public acknowledgment
    > following disclosure.

  - `$.document.aggregate_severity`

    > Provides an overall severity assessment for the report.

  - `$.document.distribution`

    > Specifies the intended sharing restrictions for the report.

>     > The `$.document.distribution.tlp.label` field SHOULD be present.

Reviewer: This is a required field. 
We could directly impose the TLP:AMBER as SHOULD requirement - then we would need a recommended test for it. Alternatively, we could split it:
6.2.39.X Usage of Lower TLP Label in Vulnerability Report
6.3.23.Y Usage of other TLP Label in Vulnerability Report than TLP:AMBER

Or we add an exception that TLP:CLEAR means that the Vulnerability is already disclosed and require proof for that in the report; happy to discuss that issue to find the best solution.

Author: Tests have been added to resolve this 6-2-39-{9,10}-{1,11} to satisfy success/failure tests.

>     > Traffic Light Protocol (TLP 2.0) markings are RECOMMENDED.

Reviewer: They are required anyway - do we then still need that sentence?

    > `TLP:AMBER` is RECOMMENDED to promote sharing on a need-to-know basis within the vendor's or
    > coordinator's organization and, where appropriate, their clients.
    > Additional information, such as `sharing_group` or `text`, MAY be provided to further clarify

>     > the publisher's intended sharing expectations.

Reviewer: see above...
```suggestion
    > the issuing party's intended sharing expectations.
```

Reviewer: We could probably add the hint that for 2 (vendor/discoverer) and 3 (vendor/coordinator/discoverer) party coordinations the usage of `sharing_group` is not needed.
What do you think?

  - `$.document.tracking`

    > Provides identifiers and version information to support coordination throughout the disclosure process.

  - `$.vulnerabilities[*].involvements`

>     > Records coordination activities between discoverers, vendors, and coordinators.

Reviewer: It might be an open ToDo to update / add `$.document.involvements` once it is there.
```suggestion
    > Records coordination activities between the parties involved in the disclosure of the specific vulnerability.
```

Author: added

  - `$.vulnerabilities[*].involvements[*].party`

    > The involvement item's `party` field SHOULD be set to `discoverer`
    > to identify the entity initiating the involvement.

  - `$.vulnerabilities[*].involvements[*].status`

    > The involvement item's `status` field MAY be set to values such as
    > `not_contacted` or `contact_attempted`
    > to indicate the coordination status between the participating parties.

  - `$.vulnerabilities[*].involvements[*].date`

    > Records when an interaction occurred.
    > Future dates MAY be used to communicate planned coordination activities,
    > including conference presentations, technical blogs, or public disclosure.
    > Future dates MAY be used to communicate or coordinate an intended public
    > disclosure date for the reported vulnerabilities.

>     > When a future date is specified, the involvement `status` is RECOMMENDED to be `open`.

Reviewer: I wonder how that translate once the new involvements thing is there. I don't have a solid suggestion yet - let's discuss some options in a meeting.

    

  - `$.vulnerabilities[*].metrics`

    > Provides vulnerability assessment metrics, such as CVSS, EPSS, SSVC,
    > or other supported scoring systems.
    > If the discoverer is aware of active exploitation, that information SHOULD be
    > communicated using the appropriate supported metric.
    > Evidence supporting known exploitation SHOULD be referenced using

>     > `$.vulnerabilities[*].references`

Reviewer: We can't have a SHOULD statement that we want to test in an informative comment. In such case, this needs to be real text.

Author: moved the SHOULD to be "should"

- The value of `$.document.category` SHALL be `csaf_superseded`.
- The elements `$.product_tree` and `$.vulnerabilities` SHALL NOT exist.

## Profile 9: Vulnerability Report

> This profile SHOULD be used to exchange an initial vulnerability report during the Coordinated Vulnerability Disclosure (CVD) process. It supports communication between discoverer, coordinator, and vendor(s). It can also be used directly between discoverer and vendor if needed.

Reviewer: It feels like it is stressing "initial" here. Is that intentional? If so, is there a special reason for that?

I wonder whether we should be agnostic first (e.g. by adding a sentence) and refer to the two different scenarios you describe here as a follow up. As the general wish is that vendors can work with discoverers without the need for a coordinator, we should consider stressing that by prioritization...

Author: yep that should initiate, not initial.

Author: should be resolved now.

> A Vulnerability Report is intended for use during the private coordination process prior to public disclosure (often referred to as the `embargo` period). It provides a structured mechanism for communicating affected products, technical details, severity assessments, and other information required to investigate, remediate, and coordinate disclosure among participating parties.

Reviewer: `embargo ` is not a key word in CSAF. Therefore, we should use another way to highlight it...

Author: Can we link to external resources, like https://certcc.github.io/CERT-Guide-to-CVD/howto/coordination/maintaining_secrecy/ to describe this. I don't see any external links in the document, perhaps that is intentional?

Author: Not sure if it is worth going deep into this topic. It is the biggest effort in the Coordination process in our experience. I have kept is very slightly mentioned only for now.

A CSAF document SHALL fulfill the following requirements to satisfy the profile "Vulnerability Report":

- The following elements MUST exist and be valid:
  - All elements required by the profile "CSAF Base".

  - `$.product_tree`

    > Lists all products referenced within the report.

  - `$.vulnerabilities`

    > Lists one or more reported vulnerabilities.

  - `$.vulnerabilities[*].notes`

>     > Provides the technical description and supporting information for each reported vulnerability.

Reviewer: Do we want to be more specific about the `category` and/or `title` that people should use? Is there specific content that we expect to be in the report?

Author: should be resolved now.

> 

Reviewer: I think we should add a requirement of `$.vulnerabilities.ids` here to have an identifier to be able to talk about even if there is no CVE assigned (yet).

Author: added

Reviewer 2: I would like to push back on making `$.vulnerabilities.ids` required (MUST) as it is not a requirement for Profile 4.

Suggest we make it recommended (SHOULD)

Author: I would like to keep it in "MUST" - if we work with `involvements` and the idea of a Case Ledger. An even internal ID field could be generated IMO. That is the only way I can think of being able talk about each vulnerability distinctly within the CSAF document. Open to discussion if I am missing something - perhaps at our 2026-08-26 1300 EDT meeting.

Reviewer: > I would like to push back on making `$.vulnerabilities.ids` required (MUST) as it is not a requirement for Profile 4.

Maybe, we should push that requirement also for profile 4? It would support our goal to have vulnerability identifiers (with a strong preference for CVE) :thinking:

Reviewer 2: I'm for consistency more so than a specific outcome, so I would be on board with that approach.

Reviewer 2: See #1559

  - `$.vulnerabilities[*].product_status`

    > Lists each product's status with respect to the reported vulnerability.

  - `$.vulnerabilities[*].product_status.known_affected`

>     > Identifies the products the discoverer has determined to be affected.

Reviewer: Shouldn't we talk about the document issuer here?

Author: should be resolved now with more text to make sure publisher understands his assertion role.

  - At least one product SHALL be listed as an item in `$.vulnerabilities[*].product_status.known_affected` for each reported vulnerability.

    > A Vulnerability Report is intended to initiate coordinated investigation of one or more known affected products.

- The value of `$.document.category` SHALL be `csaf_vulnerability_report`.

- The following elements SHOULD exist:

>   - `$.document.acknowledgements`

Reviewer: It was decided to use the form without `e`, see https://github.com/oasis-tcs/csaf/issues/142.
```suggestion
  - `$.document.acknowledgments`
```

>     > Identifies the individuals or organizations that should receive public acknowledgement for the discovery. If omitted or empty, it indicates that the discoverer wishes to remain anonymous in any public report even after the embargo period ends.

Reviewer: ```suggestion
    > Identifies the individuals or organizations that should receive public acknowledgment for the discovery. If omitted or empty, it indicates that the discoverer wishes to remain anonymous in any public report even after the embargo period ends.
```

  - `$.document.aggregate_severity`

    > Provides an overall severity assessment for the report. 

  - `$.document.distribution`

>     > Specifies the intended sharing restrictions for the report. Traffic Light Protocol (TLP 2.0) markings are RECOMMENDED.

Reviewer: As TLP is required, the question is whether we want / should provide additional information; e.g. `sharing_groups` or `text`. Maybe, we should also provide a suggested TLP label?

Author: Moved TLP to SHOULD and RECOMMENDED TLP:AMBER.  More text added on sharing_group and usage of text as well.

  - `$.document.publisher`

>     > Identifies the discoverer or organization submitting the report.

Reviewer: This is a required field; we can't have it in optional (but we could provide an example if necessary...)

Author: :) sure thing moved it up.

  - `$.document.tracking`

>     > Provides identifiers and version information to support coordination throughout the disclosure process.

Reviewer: Same here. Fill-in-instructions can live in an How-to or Appendix... (having them is definitely a good idea, same as an example.)

Author: more information on tracking IDs usage is there but concise. Not sure if an appendix exists, surely I can write a lot there

Reviewer: As it is a required field, I wonder whether we need to mention it here. Suggestion: Remove it and put aside for a how to.

  - `$.vulnerabilities[*].involvements`

    > Records coordination activities between discoverers, vendors, and coordinators.

  - `$.vulnerabilities[*].involvements[*].party`

    > The involvement item's `party` field SHOULD be set to `discoverer` to identify the entity initiating the involvement.

  - `$.vulnerabilities[*].involvements[*].status`

    > The involvement item's `status` field MAY be set to values such as `not_contacted` or `contact_attempted` to indicate the engagement history between the parties.

  - `$.vulnerabilities[*].involvements[*].date`

>     > The involvement item's `date` field records when an interaction occurred. Future dates MAY also be used to communicate planned coordination activities, including public conference presentations or blog publications. When a future date is specified, the involvement `status` is RECOMMENDED to be `open`. This future date MAY be used to negotiate an embargo period for the vulnerability.

Reviewer: Flagging this as well for @cplutte. We need to revisit this once the involvement part is settled.

Author: This a longer term work - I will work with @cplutte and @tschmidtb51  perhaps next month with a meeting focused on this.

  - `$.vulnerabilities[*].metrics`

    > Provides vulnerability assessment metrics, such as CVSS, EPSS, SSVC, or other supported scoring systems.

> 

Reviewer: Please add an explicit requirement to tell whether there is an active exploitation known...

Author: added

Reviewer: The current wording is hard to test as we need to check every supported scoring system. What about binding it to SSVC with the concrete value?
And adding the first_known_exploitation_dates as additional MAY (=> informative test; not profile specific).

Author: SSVC is the required metric for Exploitation now.

- The value of `$.document.category` SHALL be `csaf_superseded`.
- The elements `$.product_tree` and `$.vulnerabilities` SHALL NOT exist.

## Profile 9: Vulnerability Report

This profile SHOULD be used to exchange a vulnerability report during the Coordinated Vulnerability Disclosure (CVD) process.
It supports communication between discoverers, vendors, and, where applicable, coordinators prior to public disclosure.
A Vulnerability Report is intended for use during the private coordination process prior to public disclosure.
It provides a structured mechanism for exchanging information throughout the coordinated disclosure process among the participating parties.
Information such as affected products, technical details, severity assessments, and supporting evidence can be
structured and exchanged privately using this profile to facilitate coordination.

> A CSAF document SHALL fulfill the following requirements to satisfy the profile "Vulnerability Report":

Reviewer 3: Maybe use MUST instead of SHALL? Else we start to confuse implementers. Also, it may be a cheap shot to create a set before listing it. Especially when the requirements are ionterspersed with many informational statments.

Reviewer: The language seems to be copied / consistent with the other profiles. Given the preference, I suggest to either
a) change both to SHALL and fix other profiles in an editorial PR or
b) keep it consistent and fix all later with one PR:

Author: I second we leave this SHALL language for now
